### PR TITLE
Wallet design review fixes

### LINKED ISSUE #21221: Wallet - Account screen issues
# Design Feedback 2.30

## [Wallet] Account screen issues

- **Currency** [[1](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934563295)] should be set to € by default
- **Bridge icon** [[2](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934562576)] still needs to be updated
- **Metrics** [[3](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934563934)] should be removed from the _Token list_ keeping the same alignment for the balance

#### Expected behavior

Screenshot 2024-09-09 alle 10 48 35

#### Actual behavior

Screenshot 2024-09-09 alle 10 48 55

### Reproduction

1. Open the Wallet
2. Select an empty account

### Additional Information

- Operating System: Android, iOS
- Devices in the screenshots: iPhone 13

### LINKED ISSUE #21223: Remove dApp connections button from Bridge screen
# Design Feedback 2.30

## dApp connections icon button shouldn't be in Bridge screen
the dApp icon should only be present in Wallet account screen, not during the bridging flow since it doesn't make sense from a UX perspective.

Figma review [here](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#941531183)

Screenshot 2024-09-09 alle 11 48 52

### Reproduction

1. Open a Wallet account
2. Tap on Bridge

### Additional Information

- Operating System: Android, iOS
- Devices in the screenshots: iPhone 13

### LINKED ISSUE #21224: Wallet - Bridge screen issues
# Design Feedback 2.30

## [Wallet] Bridge screen issues

- **Currency** [[1](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#938001507)] should be set to € by default
- **Icon in the button** [[2](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#936326581)] shouldn't be there
- **Button colour** [[3](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#936326581)] should be the Wallet account colour

------------------------------
### Android only

- **Currency** [[4](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#937861419)] in the input field is misaligned 

#### Expected behavior

Screenshot 2024-09-09 alle 12 51 36

#### Actual behavior

Screenshot 2024-09-09 alle 12 52 00

### Reproduction

1. Open Wallet
2. Tap Bridge

### Additional Information

- Operating System: Android, iOS
- Devices in the screenshots: iPhone 13, Realme 10

### PR DESCRIPTION
**Pull request by clauxx**
fixes #21223 
fixes #21224 
fixes #21221 

### Summary

Some of the design issues have been already fixed it seems, as I couldn't reproduce them, so this is what was fixed here:
1. hide the "dApp connections" button from the header, unless we're on the account screen
2. hide the "token metrics" (price change indicator) if token has no balance

Not sure about using EUR as the default currency, as that would change it on desktop as well. 
@shivekkhurana is there a decision on using EUR as the default currency?

### Screenshots

<br><br/>

<br><br/>

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/common/account_switcher/view.cljs [hunk 1]
```diff
@@ -16,15 +16,17 @@
                                  [select-account/view params])}
     nil))
 
-(defn- on-dapps-press
+(defn- on-switcher-press
   [switcher-type params]
   (rf/dispatch [:show-bottom-sheet (get-bottom-sheet-args switcher-type params)]))
```
Comment (clauxx) [inline src/status_im/contexts/wallet/common/account_switcher/view.cljs:21]: renamed since it has no relation to dapps

**Review by mohsen-ghafouri — APPROVED** [review]

**Comment by shivekkhurana** [comment]
> @shivekkhurana is there a decision on using EUR as the default currency?
I think at the moment, we are forced to. Because we don't have a currency switcher on mobile.

@smohamedjavid @xAlisher Do you know if EURO is the default by choice or by force ?

**Comment by clauxx** [comment]
> > @shivekkhurana is there a decision on using EUR as the default currency?
> > I think at the moment, we are forced to. Because we don't have a currency switcher on mobile.
> 
> @smohamedjavid @xAlisher Do you know if EURO is the default by choice or by force ?

@shivekkhurana we do have a currency switcher on mobile though :thinking:

**Comment by smohamedjavid** [comment]
> > @shivekkhurana is there a decision on using EUR as the default currency?
> > I think at the moment, we are forced to. Because we don't have a currency switcher on mobile.
> 
> @smohamedjavid @xAlisher Do you know if EURO is the default by choice or by force ?

I believe we have been using USD as the default currency for both clients. If I recall correctly, there was a discussion very long ago to make EUR the default, but it's not done yet (not sure it's captured somewhere).

Additionally, We also rely on USD, especially for rounding off decimals in the token balance 
https://github.com/status-im/status-mobile/blob/9862abb7ebaac387652b9f3ff263b96198a3679f/src/status_im/contexts/wallet/common/utils.cljs#L113-L127

**Comment by smohamedjavid** [comment]
> @shivekkhurana we do have a currency switcher on mobile though 🤔

@clauxx - Yes, we do have. It can be done through `Profile > Language and currency > Currency`. If the user changes currency on any of the clients, the synced devices will be updated too.

**Review by smohamedjavid — APPROVED** [review]

File: src/status_im/contexts/wallet/common/utils.cljs [hunk 2]
```diff
@@ -275,7 +275,7 @@
     {:token               (:symbol token)
      :token-name          (:name token)
      :state               :default
-     :metrics?            true
+     :metrics?            (money/has-funds? balance)
```
Comment (vkjr) [inline src/status_im/contexts/wallet/common/utils.cljs]: When I'm reading `has-funds?` I have a feeling that it is some calculation on complex big structure whereas it is simply a comparison to zero :) I'd suggest to rename `has-funds?`-> `greater-than-zero?` to leave `utils.money` on lower level of abstraction and to be very precise of what function does, wdyt?

**Review by vkjr — APPROVED** [review]

File: src/status_im/contexts/wallet/common/utils.cljs (hunk 2 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/common/utils.cljs]: Good point! Renamed it to `above-zero?` (basically the same but shorter) and replaced the usage of `greater-than` where the value is compared to `0` to use this function instead.

**Comment by pavloburykh** [comment]
@clauxx thanks for the fixes. 

Speaking of https://github.com/status-im/status-mobile/issues/21224 I see that 2 reported points of the issue are still valid:

> Currency [[1](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#938001507)] should be set to € by default
> Icon in the button [[2](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#936326581)] shouldn't be there

photo_2024-11-04 16 51 45

I see you have already raised a question about default euros, so we need to decide what to do with that. Do we need to fix it? if yes, then where do we want to fix it: in scope of this PR or log a separate issue?

**Comment by pavloburykh** [comment]
Similar situation with https://github.com/status-im/status-mobile/issues/21221. Two points are still valid: 

>Currency [[1](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934563295)] should be set to € by default
>Bridge icon [[2](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934562576)] still needs to be updated

**Comment by clauxx** [comment]
> Similar situation with #21221. Two points are still valid:
> 
> > Currency [[1](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934563295)] should be set to € by default
> > Bridge icon [[2](https://www.figma.com/design/oznddnBnDbLlLQIaACYNuj?node-id=2224-52505#934562576)] still needs to be updated

@pavloburykh thanks for having a look!

Regarding the currency, I think it's best to have a separate issue for it, as it doesn't seem like there's a clear decision on it.

The bridge icon was fixed [here](https://github.com/status-im/status-mobile/pull/21514), should be here now as well.

Removed the icon in the button :+1:

**Comment by pavloburykh** [comment]
Thanks for the PR @clauxx. LGTM!

> Regarding the currency, I think it's best to have a separate issue for it, as it doesn't seem like there's a clear decision on it.

I have logged a separate issue regarding default currency https://github.com/status-im/status-mobile/issues/21582. 

@Francesca-G could you please review current PR and tell if it is okay from design perspective. Following issues are fixed:
https://github.com/status-im/status-mobile/issues/21223
https://github.com/status-im/status-mobile/issues/21224
https://github.com/status-im/status-mobile/issues/21221

Mind that default EUR currency has moved to separate [issue](https://github.com/status-im/status-mobile/issues/21582) as we do not have clear decision about implementation yet. 

@clauxx once PR is approved by @Francesca-G it will be ready for merge.

**Review by Francesca-G — APPROVED** [review]
Looks good to me 👍

**Comment by pavloburykh** [comment]
> Looks good to me 👍

Thank you @Francesca-G.

@clauxx PR is ready for merge now.

**Review by alwx — APPROVED** [review]

**Pull request merged**